pkg/v1/daemon/image.go

```diff
@@ -99,6 +100,9 @@ func Image(ref name.Reference, options ...Option) (v1.Image, error) {
 }
 
 func (i *image) initialize() error {
+	i.mu.Lock()
```

Comment:
Instead of a mutex, could we use a [`sync.Once`](https://pkg.go.dev/sync#Once)? This would call `tarball.Image` only the first time `initialize` is called, taking into account multiple concurrent calls.

Reply:
Sure. As described in the description, we need to keep the err. Is it ok with you?

>Note that we need to keep err in the image struct like imageOpener.

Reply:
Fixed https://github.com/google/go-containerregistry/pull/1126/commits/3eac11ecf1a82f386ecf187e58c802eb8b27a3f1